packages/transactions/src/signer.ts

```diff
@@ -19,6 +19,26 @@ export class TransactionSigner {
     this.originDone = false;
     this.checkOversign = true;
     this.checkOverlap = true;
+
+    // If multi-sig spending condition exists, iterate over
+    // auth fields and reconstruct sigHash
+    let spendingCondition = transaction.auth.spendingCondition;
+    if (spendingCondition && !isSingleSig(spendingCondition)) {
+      spendingCondition.fields.forEach(field => {
+        if (field.contents.type === StacksMessageType.MessageSignature) {
+          const signature = field.contents;
+          const nextVerify = nextVerification(
+            this.sigHash,
+            transaction.auth.authType!,
+            spendingCondition!.fee,
+            spendingCondition!.nonce,
```

Comment:
Maybe better to explicitly check for these optionals and throw a useful error if they are undefined?

Reply:
went ahead and implemented the `Authorization` refactor I mentioned above: https://github.com/blockstack/stacks.js/pull/1082

Was able to get rid of these null checks (in that PR) as a result